src/Model.jl

```diff
@@ -98,6 +103,7 @@ function Model(lattice::AbstractMatrix{T}, atoms=Element[], positions=Vec3{T}[];
                spin_polarization=default_spin_polarization(magnetic_moments),
                symmetries=default_symmetries(lattice, atoms, positions, magnetic_moments,
                                              spin_polarization, terms),
+               system=construct_system(lattice, atoms, positions, magnetic_moments),
```

Comment:
Hm, this allows passing both a lattice/atoms/positions and a system without enforcing consistency. Do you really need it as an input arg?